### PR TITLE
hubble: Update uint size in flow proto

### LINKED ISSUE #158: Revisit integer sizes in protobuf definitions
Some of our integer sizes in the protobuf definitions are too large. If we can afford a breaking change, we should resize them to appropriate sizes.

Example:
https://github.com/cilium/hubble/blob/23ae0b9c2a2d399789faf6189db8820063b608c0/api/v1/flow/flow.proto#L114
Cilium numeric security identities are 32 bit. We should use a uint32 here.

Unfortunately protobuf does not have a `uint16` type, which would be appropriate for things like port numbers or `pb.Endpoint.ID`. We will have to keep using `uint32` there.

### PR DESCRIPTION
**Pull request by matej-g**
Size of some fields in flow.proto is unnecessarily
large (uint64), whereas for such fields
(e.g. endpoint ID or identity) a smaller type
is more appropriate.

This PR is a clean-up  to update the scalar
value types of these fields. Includes some
changes to related Hubble packages (parsers
and filters) and minor test adjustments.

Fixes: cilium/hubble#158 (this is an issue originally filed in Hubble repository, before the code was moved over to Cilium)

Signed-off-by: Matej Gera <matejgera@gmail.com>

```release-note
hubble: Update uint size in flow proto
```

### CODE REVIEW

**Comment by gandro** [comment]
test-me-please

**Comment by gandro** [comment]
Context for this PR for other reviewers: Back when Hubble was running in its own process, we used to obtain metadata via the Swagger-based Cilium API, which uses `int64` for any integer types. This size was then used across the code base and the protobuf definitions, even though the actual endpoint ids are 16 bit, and the security identities are 32 bit in the datapath. 

Changing the integer sizes of the protobuf definitions does not break backwards-compatibility: https://developers.google.com/protocol-buffers/docs/proto3#updating

File: pkg/hubble/parser/seven/parser.go [hunk 1]
```diff
@@ -280,8 +280,8 @@ func decodeEndpoint(endpoint accesslog.EndpointInfo, namespace, podName string)
 	labels := endpoint.Labels
 	sort.Strings(labels)
 	return &pb.Endpoint{
-		ID:        endpoint.ID,
-		Identity:  endpoint.Identity,
+		ID:        uint32(endpoint.ID),
```
Comment (rolinh) [inline pkg/hubble/parser/seven/parser.go:283]: Does anyone know why `pkg.proxy.accesslog.EndpointInfo.[ID,Identity]` are `uint64` ? These type casts don't look safe to me.

Comment (gandro) [reply pkg/hubble/parser/seven/parser.go:283]: See https://github.com/cilium/cilium/pull/11161#issuecomment-619830199 for the rationale - the values will never exceed 32 bit (or 16 bit in this case).

Comment (rolinh) [reply pkg/hubble/parser/seven/parser.go:283]: But these two instances are from Cilium's proxy package hence not something that falls into your description, no?

Comment (gandro) [reply pkg/hubble/parser/seven/parser.go:283]: True, but they suffered a similar fate: `EndpointInfo` was intended for JSON serialization (see  https://github.com/cilium/cilium/pull/964), therefore I assume the type definitions were generated from some JSON-to-Go-struct tool, which picked `int64` for the type.

I've checked all initialization of `EndpointInfo.ID` and they all take the value taken from `GetID` here (also added by the above PR) which casts a 16 bit integer to 64 bit
https://github.com/cilium/cilium/blob/dd67d2d37f1a07ed5eeaff68cffd09212de62910/pkg/endpoint/endpoint.go#L429

So I don't see any safety issue here.

**Review by gandro — APPROVED** [review]
Awesome! Thanks a lot for taking care of this.

We should also review the uses of `uint64` for the endpoint id and identiy in other parts of the code, i.e. the getters in https://github.com/cilium/cilium/blob/86045f79ad6c3aeb104a0092d48c3691130b1699/pkg/hubble/parser/getters/getters.go also unnecessarily take uint64 parameters. But let's do that in a follow-up PR.

**Review by rolinh — APPROVED** [review]

**Comment by gandro** [comment]
test-gke

**Review by aanm — APPROVED** [review]
@gandro GKE is not required to pass in PRs at this moment.

**Comment by matej-g** [comment]
> We should also review the uses of `uint64` for the endpoint id and identiy in other parts of the code, i.e. the getters in https://github.com/cilium/cilium/blob/86045f79ad6c3aeb104a0092d48c3691130b1699/pkg/hubble/parser/getters/getters.go also unnecessarily take uint64 parameters. But let's do that in a follow-up PR.

Good call about the getters, wasn't sure whether I should include it or not as it also requires change in daemon. I will open an issue for that + PR.

**Pull request merged**